### PR TITLE
batch kwarg generator rename and legacy python hacks removed

### PR DESCRIPTION
**Pull request by Aylr**
## What's Here

- A small bugfix where `DataSource.list_batch_kwarg_generators` was returning a dictionary with empty values instead of an empty list.
- A major rename of all things generator to batch kwarg generator
- A substantial slash and burn of legacy python hacks and workarounds

**Note this is a breaking PR**
﻿

### CODE REVIEW

**Review by ayirplm — COMMENTED** [review]
first pass items -

What’s at index 0?
db_choices = [str(x) for x in list(range(1, 1 + len(SupportedDatabases)))]
great_expectations/cli/datasource.py

great_expectations/cli/datasource.py
 msg_prompt_select_generator = "Select batch kwarggenerator"
Maybe keep the underscore in Select batch ....

max_data_assets=20,
def profile_datasource(

Is this validated for 20 data assets?

Specify the one you want the profiler to use in batch_kwarg_generator_name argument.
In some cases, comment has underscore and not in others….good to be consistent

great_expectations/cli/tap.py 
Case consistency,Beta, BETA


Line 1281, typo
 DataContexts use data sources you're already familiar with. BathKwargGenerators
great_expectations/data_context/data_context.py

Line 1292
 The BathKwargGenerator knows ho


great_expectations/datasource/sqlalchemy_datasource.py 
from great_expectations.types.configurations import ClassConfigSchema, classConfigSchema

Very long path intentional, if someone’s running this on a raspi?
"path": "/private/var/folders/_t/psczkmjd69vf9jz0bblzlzww0000gn/T/pytest-of-taylor/pytest-1812/filesystem_csv0/f1.csv",
tests/cli/tap.py

a_new_new_suite_name is like naming dogs Jade1,Jade2,Jade3

File: great_expectations/cli/cli_logging.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 warnings.filterwarnings("ignore")
 
 logging.getLogger(
-    "great_expectations.datasource.generator.in_memory_generator"
+    "great_expectations.datasource.batch_kwarg_generator.in_memory_batch_kwarg_generator"
```
Comment (jcampbell) [inline great_expectations/cli/cli_logging.py]: In Memory Generator has already been removed, the CLI shouldn't try to touch its (nonexistent) logger

File: great_expectations/cli/datasource.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
     SparkDFDatasource,
     SqlAlchemyDatasource,
 )
-from great_expectations.datasource.generator import ManualBatchKwargsGenerator
-from great_expectations.datasource.generator.table_generator import (
+from great_expectations.datasource.batch_kwarg_generator import ManualBatchKwargsGenerator
+from great_expectations.datasource.batch_kwarg_generator.table_batch_kwarg_generator import (
```
Comment (jcampbell) [inline great_expectations/cli/datasource.py]: pluralize ("batch_kwargs generator")

File: great_expectations/cli/datasource.py [hunk 3]
```diff
@@ -39,12 +39,6 @@
 
 logger = logging.getLogger(__name__)
 
-# FIXME: This prevents us from seeing a huge stack of these messages in python 2. We'll need to fix that later.
-# tests/test_cli.py::test_cli_profile_with_datasource_arg
-#   /Users/abe/Documents/superconductive/tools/great_expectations/tests/test_cli.py:294: Warning: Click detected the use of the unicode_literals __future__ import.  This is heavily discouraged because it can introduce subtle bugs in your code.  You should instead use explicit u"" literals for your unicode strings.  For more information see https://click.palletsprojects.com/python3/
-#     cli, ["profile", "my_datasource", "-d", project_root_dir])
-click.disable_unicode_literals_warning = True
-
```
Comment (jcampbell) [inline great_expectations/cli/datasource.py:52]: <3 <3 <3

File: great_expectations/cli/datasource.py [hunk 4]
```diff
@@ -123,10 +117,10 @@ def datasource_list(directory):
 @datasource.command(name="profile")
 @click.argument('datasource', default=None, required=False)
 @click.option(
-    "--generator-name",
+    "--batch-kwarg-generator-name",
```
Comment (jcampbell) [inline great_expectations/cli/datasource.py]: pluralize

**Review by jcampbell — CHANGES_REQUESTED** [review]
Woohoo, nice cleanup.

Request a consistent rename of batch_kwarg_generator -> batch_kwargs_generator for consistency with the actual class names

File: great_expectations/cli/cli_logging.py (hunk 1 above)
Comment (Aylr) [reply great_expectations/cli/cli_logging.py]: Fixed

File: great_expectations/cli/datasource.py (hunk 4 above)
Comment (Aylr) [reply great_expectations/cli/datasource.py]: fixed

**Comment by jcampbell** [comment]
Most uses appeared not to have been pluralized...

**Comment by jcampbell** [comment]
I rebased onto feature/usage_statistics. This is now LGTM (updating review).

**Review by jcampbell — APPROVED** [review]

**Pull request merged**